crates/burn-jit/src/kernel/conv/conv2d/col2im.rs

```diff
@@ -0,0 +1,289 @@
+        padding_out_h,
+        dilation_h,
+        input_h,
+    );
+    let im_w = calculate_conv_transpose_output_size(
+        kernel_w,
+        stride_w,
+        padding_w,
+        padding_out_w,
+        dilation_w,
+        input_w,
+    );
+    let im_channels = im_ch_per_group * groups;
+
+    let batches_per_run = batches_per_run(batch_size, input_h, input_w);
+    let col_shape_0 = im_ch_per_group * kernel_h * kernel_w;
+
+    let weight = reshape(
+        weight.clone(),
+        Shape::new([groups, input_ch_per_group, col_shape_0]),
+    );
+    let weight = into_contiguous(swap_dims(weight, 1, 2));
+
+    let mut image = if batches_per_run != batch_size {
+        let runs = batch_size / batches_per_run;
+
+        let im_shape = Shape::new([runs, batches_per_run, im_channels, im_h, im_w]);
+        let mut image = empty_device(input.client.clone(), input.device.clone(), im_shape);
+
+        let input_shape = Shape::new([runs, batches_per_run, input_channels, input_h, input_w]);
+        let input = reshape(input, input_shape);
+        let input_shape_run = Shape::new([batches_per_run, input_channels, input_h, input_w]);
+        let im_shape_run = Shape::new([1, batches_per_run, im_channels, im_h, im_w]);
+
+        for run in 0..runs {
+            let input = JitBackend::<R, E, I>::float_narrow(input.clone(), 0, run, 1);
+            let input = reshape(input, input_shape_run.clone());
+            let image_run = execute::<R, E, I>(
+                input,
+                weight.clone(),
+                options.clone(),
+                im_ch_per_group,
+                im_h,
+                im_w,
+                kernel_h,
+                kernel_w,
+            );
+            let image_run = reshape(image_run, im_shape_run.clone());
+            image = JitBackend::<R, E, I>::float_slice_assign(
+                image,
+                [
+                    run..run + 1,
+                    0..batches_per_run,
+                    0..im_channels,
+                    0..im_h,
+                    0..im_w,
+                ],
+                image_run,
+            )
+        }
+        reshape(image, Shape::new([batch_size, im_channels, im_h, im_w]))
+    } else {
+        execute::<R, E, I>(
+            input,
+            weight,
+            options,
+            im_ch_per_group,
+            im_h,
+            im_w,
+            kernel_h,
+            kernel_w,
+        )
+    };
+
+    if let Some(bias) = bias {
+        let ones = ones_device(
+            client.clone(),
+            device.clone(),
+            Shape::new([1, batch_size * im_h * im_w]),
+        );
+        let bias = reshape(bias, Shape::new([im_channels, 1]));
+        let bias = JitBackend::<R, E, I>::float_matmul(bias, ones);
+        let bias = reshape(bias, Shape::new([im_channels, batch_size, im_h, im_w]));
+        let bias = swap_dims(bias, 0, 1);
+        image = JitBackend::<R, E, I>::float_add(image, bias);
```

Comment:
Why not start the convolution with bias as the input?

Reply:
Because this is faster. The columns matrix contains a lot of duplicates, so you would end up adding the bias several times for each pixel. Adding it at the end takes far fewer operations.